source/Private/Test-FabricApiResponse.ps1

```diff
@@ -0,0 +1,129 @@
+function Test-FabricApiResponse {
+
+    <#
+.SYNOPSIS
+    Tests the response from a Fabric API call and handles long-running operations.
+
+.DESCRIPTION
+    Tests the response from a Fabric API call and handles long-running operations. It checks the status code and processes the response accordingly.
+
+.PARAMETER Response
+    The response body from the API call.
+
+.PARAMETER ObjectIdOrName
+    The name or ID of the resource being operated.
+
+.PARAMETER TypeName
+    The type of resource being operated (default: 'Fabric Item').
+
+.PARAMETER NoWait
+    If specified, the function will not wait for the operation to complete and will return immediately.
+
+.EXAMPLE
+    Test-FabricApiResponse -statusCode 201 -response $response -responseHeader $header -Name "MyResource" -typeName "Fabric Item"
+
+    Handles the response from a Fabric API call with a 201 status code, indicating successful creation of a resource.
+
+.EXAMPLE
+    Test-FabricApiResponse -statusCode 202 -response $response -responseHeader $header -Name "MyResource" -typeName "Fabric Item"
+
+    Handles the response from a Fabric API call with a 202 status code, indicating that the request has been accepted for processing.
+
+.NOTES
+    - This function is designed to be used within the context of a Fabric API client.
+    - It requires the `Write-Message` function to log messages at different levels (Info, Debug, Error).
```

Comment:
Lets not use write-message here - Better more friendly options are available and this will save adding more technical debt and leave the function better than we find it. I would suggest that you return to default PowerShell Write-* commands until the proffered logging option is added